src/net.h

```diff
@@ -1115,11 +1120,16 @@ class CNode
 
     void PushAddress(const CAddress& _addr, FastRandomContext &insecure_rand)
     {
+        // Whether the peer supports the address in `_addr`. For example,
+        // nodes that do not implement BIP155 cannot receive Tor v3 addresses
+        // because they require ADDRv2 (BIP155) encoding.
+        const bool addr_format_supported = m_wants_addrv2 || _addr.IsAddrV1Compatible();
```

Comment:
`PushAddress()` is called from 5 places. We don't want to duplicate the same code in 5 places.

Reply:
`PushAddress()` is called from 5 places. We don't want to duplicate the same code in 5 places.

Reply:
Well another alternative is a new method `IsAddrUnderstood` on `CNode` making the call to `PushAddress()` conditional on it results. It can be done later, not a real concern.

Reply:
`PushAddress()` and all of the addr relay data will eventually move into net processing. See the commits in https://github.com/jnewbery/bitcoin/commits/2020-06-cs-main-split. That work obviously shouldn't hold up new features like BIP 155 and I'll continue to rebase those commits on master as things change. #19829 and #19910 are the next PRs to review if you'd like to help.